medusa/post_processor.py

```diff
@@ -302,7 +302,7 @@ def _delete(self, files, associated_files=False):
 
         # also delete associated files, works only for 1 file
         if associated_files and len(files) == 1:
-            file_list += self.list_associated_files(files, subfolders=True)
+            file_list += self.list_associated_files(files[0], subfolders=True)
```

Comment:
Isn't this the reason of folder not being deleted in root folder as its only deletes one associated file ?

`# also delete associated files, works only for 1 file`